comp/dogstatsd/server/server.go

```diff
@@ -144,6 +146,29 @@ type server struct {
 	enrichConfig enrichConfig
 }
 
+func init() {
+	expvarcollector.RegisterExpvarReport("dogstatsdStats", func() (interface{}, error) {
```

Comment:
I think we really have to try hard not to have any of these in the `init()` methods. Part of the componentization effort is all about removing them from there. We won't be able to handle all cases without `init()` but we probably need to find alternatives where possible.

Reply:
Minor nit suggestion:  `RegisterExpvarReport` -> `RegisterExpvarCallback`

Reply:
> I think we really have to try hard not to have any of these in the init() methods.

I completely agree with that statement. We need to balance out the best temporary strategy for these changes. Spending a lot of time creating a new pattern to remove the `init()` functions might be overkill if it won't be reused later when the status and subsequent components are implemented. 

On the contrary, reusing already existing `init()` functions or adding a few `init()` might be a good compromise, knowing that they would get removed once each component is created.